apps/redi-connect/src/pages/app/find-a-mentor/FindAMentor.tsx

```diff
@@ -200,11 +201,28 @@ const FindAMentor = () => {
   const menteeHasAnActiveMatch =
     profile?.userType === 'MENTEE' &&
     profile?.mentorshipMatches.length > 0 &&
-    profile?.mentorshipMatches?.[0].status === 'ACCEPTED'
+    profile?.mentorshipMatches?.some(match =>
+      match.status === MentorshipMatchStatus.Accepted
+    )
 
   if (menteeHasAnActiveMatch) {
-    const matchId = profile?.mentorshipMatches?.[0].id
-    history.replace(`/app/mentorships/${matchId}`)
+    const activeMentorshipMatch = 
+      profile.mentorshipMatches.find(match =>
+        match.status === MentorshipMatchStatus.Accepted
+      )
```

Comment:
Not a significant issue, as profile.mentorshipMatches can never be a huge list anyway, but instead of two iterations, you could use find to get the accepted mentorship match in the beginning, use it also in `menteeHasAnActiveMatch` calculation, and then directly here if the condition is true.

Something like this:

```js
const activeMentorshipMatch = profile.mentorshipMatches.find(...);

const menteeHasAnActiveMatch = ... && !!activeMentorshipMatch;

if (menteeHasAnActiveMatch) {
  // use activeMentorshipMatch directly here.
}

```

Reply:
Agree - pushed a fix